gcp-auth-extension/src/test/java/io/opentelemetry/contrib/gcp/auth/GcpAuthCustomizerProviderTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.gcp.auth;
+
+import static io.opentelemetry.contrib.gcp.auth.GcpAuthCustomizerProvider.SIGNAL_TARGET_WARNING_YAML_FIX_SUGGESTION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfiguration;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+class GcpAuthCustomizerProviderTest {
+
+  @Test
+  void declarativeConfig() throws IOException {
+    String yaml =
+        "file_format: 0.4\n"
+            + "tracer_provider:\n"
+            + "  processors:\n"
+            + "    - simple:\n"
+            + "        exporter:\n"
+            + "          otlp_http:\n"
+            + "meter_provider:\n"
+            + "  readers:\n"
+            + "    - periodic:\n"
+            + "        exporter:\n"
+            + "          otlp_http:\n"
+            + "instrumentation/development:\n"
+            + "  java:\n"
+            + "    google:\n"
+            + "      cloud:\n"
+            + "        project: p\n"
+            + "        quota:\n"
+            + "          project: qp\n"
+            + "      otel:\n"
+            + "        auth:\n"
+            + "          target:\n"
+            + "            signals: [metrics, traces]\n";
+
+    OpenTelemetryConfigurationModel model =
+        DeclarativeConfiguration.parse(
+            new ByteArrayInputStream(yaml.getBytes(StandardCharsets.UTF_8)));
+    ConfigProperties properties = GcpAuthCustomizerProvider.getConfigProperties(model);
+
+    assertThat(GcpAuthAutoConfigurationCustomizerProvider.targetSignals(properties))
+        .containsExactly("metrics", "traces");
+    assertThat(GcpAuthAutoConfigurationCustomizerProvider.getProjectId(properties)).isEqualTo("p");
+    assertThat(GcpAuthAutoConfigurationCustomizerProvider.getQuotaProjectId(properties))
+        .contains("qp");
+
+    GoogleCredentials credentials = mock(GoogleCredentials.class);
+    when(credentials.getRequestMetadata())
+        .thenReturn(
+            Collections.singletonMap("x-goog-user-project", Collections.singletonList("qp")));
+
+    GcpAuthCustomizerProvider.customize(model, credentials, properties);
+
+    String header =
```

Comment:
An additional header that is important to check here is the auth header. 
`[name=Authorization, value=bearer fake-credential-token]`. 

Sample setup:
```java
mockedRequestMetadata =
        ImmutableMap.of(
              "Authorization",
              Collections.singletonList("Bearer fake-access-token"),
              QUOTA_USER_PROJECT_HEADER,
              Collections.singletonList("qp"));

Mockito.when(mockedGoogleCredentials.getRequestMetadata()).thenReturn(mockedRequestMetadata);
```

PS: I can update the testing later as well.